### PR TITLE
Bump Go to 1.8.4

### PR DESCRIPTION
**Pull request by thaJeztah**
Bumps the Go version used to 1.8.4, which contains security fixes; https://groups.google.com/forum/#!topic/golang-announce/1hZYiemnkdE

### CODE REVIEW

File: Dockerfile.e2e [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ## Step 1: Build tests
-FROM golang:1.8.3-alpine3.6 as builder
+FROM golang:1.8.4-alpine3.6 as builder
```
Comment (thaJeztah) [inline Dockerfile.e2e]: This will currently fail, because updated docker images are not yet live on Docker Hub

/cc @tianon

**Comment by fntlnz** [comment]
Worth mentioning in https://github.com/moby/moby/blob/master/project/PACKAGERS.md#build-dependencies ?

**Comment by thaJeztah** [comment]
Most of those will probably be removed at some point, with the packaging scripts for _docker_ now being in the https://github.com/docker/docker-ce-packaging repository. I know that backward compatibility with Go 1.6 was attempted to a while to allow building on  distros that don't ship newer versions, but I'm not even sure if that still works.

File: Dockerfile.e2e (hunk 1 above)
Comment (tianon) [reply Dockerfile.e2e]: Should be live now!

Comment (thaJeztah) [reply Dockerfile.e2e]: Image looks to be up now:

```
1.8.4-alpine3.6: Pulling from library/golang
88286f41530e: Already exists
3f27971ac235: Pull complete
102255e55471: Pull complete
baa88d0dbe5f: Pull complete
38ab85760b4b: Pull complete
fa44aa219fdc: Pull complete
Digest: sha256:c21963f3a84d7ffb93a2fc60c8426db810abc372e95995d933c06170b897cba4
Status: Downloaded newer image for golang:1.8.4-alpine3.6
```

**Comment by fntlnz** [comment]
I suspected this, why not remote all this kind of things? I think that can be potentially confusing for users.

**Comment by thaJeztah** [comment]
I have a branch that I was working on for that, but there's still some parts that are not included in the ce-packaging repository for which we need a home

**Comment by fntlnz** [comment]
Worth having a temporary home? Like how we did for the tool?

**Review by vdemeester — APPROVED** [review]
LGTM 🐮

**Review by yongtang — APPROVED** [review]
LGTM

**Pull request merged**